src/widgets/Console.ts

```diff
@@ -0,0 +1,91 @@
+import { v, w } from '@dojo/widget-core/d';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import { WidgetProperties } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, ThemedProperties, theme } from '@dojo/widget-core/mixins/Themed';
+import { stringify } from '../util';
+import ActionBar, { ActionBarButton } from './ActionBar';
+import * as css from '../styles/console.m.css';
+
+export enum ConsoleMessageType {
+	Log = 'log',
+	Error = 'error',
+	Info = 'info',
+	Warning = 'warn'
+}
+
+export interface ConsoleMessage {
+	type: ConsoleMessageType;
+	message: any | any[];
+	filename?: string;
+	lineNumber?: number;
+}
+
+export interface ConsoleProperties extends WidgetProperties, ThemedProperties {
```

Comment:
I have it a few other places (which I realised were wrong, but it should only be `ThemedProperties` as `ThemedProperties` already extends `WidgetProperties`.